### ISSUE TITLE
Suggest (or x y) instead of (if x x y)

### ISSUE THREAD

**Issue by pmonks** [clj-kondo/clj-kondo#2062]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**
Sometimes, especially after an extended break from Clojure, I use `(if x x y)` when `(or x y)` is more idiomatic.

**Describe the solution you'd like**
It would be great if kondo detected the former pattern and suggested the latter pattern when it sees it in code.

**Describe alternatives you've considered**
[Eastwood](https://github.com/jonase/eastwood) also doesn't detect this anti-pattern.

**Additional context**
None.

**Comment by borkdude** [comment]
`or` expands into this:

```
(let* [or__5581__auto__ 1] (if or__5581__auto__ or__5581__auto__ (clojure.core/or 1)))
```

Not sure which one I like better :P

**Comment by borkdude** [comment]
But seriously though a PR is welcome if anybody is up for it.

**Comment by pmonks** [comment]
Bump.

**Comment by pmonks** [comment]
This is still a valuable linting rule.

**Comment by pmonks** [comment]
Bump.
